From 3723e163041a6b0801b50e5a679f176c2747b6e5 Mon Sep 17 00:00:00 2001
From: Jon Lin <jon.lin@rock-chips.com>
Date: Fri, 3 Jul 2020 19:25:35 +0800
Subject: [PATCH] Revert "spi: rockchip: transfer SPI slave by CPU loop"

This reverts commit 846019f5fa482ddb8b70b28decb7ec815a5fa4b6 which is
commit 810923f3bf06c11022b7bb370dd3130a956a222e upstream.

Upstream add wait_for_completion_interruptible to hold SPI transfer
in SPI slave work, so it support SPI slave IRQ or DMA transfer.

Signed-off-by: Jon Lin <jon.lin@rock-chips.com>
Change-Id: Ie0202b0d2d7b9ff7bfc462587c2fc643155549f1
---
 drivers/spi/spi-rockchip.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/spi/spi-rockchip.c b/drivers/spi/spi-rockchip.c
index 1626800ef7c4..f850052ca83b 100644
--- a/drivers/spi/spi-rockchip.c
+++ b/drivers/spi/spi-rockchip.c
@@ -821,8 +821,7 @@ static int rockchip_spi_probe(struct platform_device *pdev)
 		rs->dma_rx.ch = NULL;
 	}
 
-	if (rs->dma_tx.ch && rs->dma_rx.ch &&
-	    !spi_controller_is_slave(master)) {
+	if (rs->dma_tx.ch && rs->dma_rx.ch) {
 		dma_get_slave_caps(rs->dma_rx.ch, &(rs->dma_caps));
 		rs->dma_tx.addr = (dma_addr_t)(mem->start + ROCKCHIP_SPI_TXDR);
 		rs->dma_rx.addr = (dma_addr_t)(mem->start + ROCKCHIP_SPI_RXDR);
-- 
2.35.3

